### ISSUE TITLE
Hang with df -h

### ISSUE THREAD

**Issue by keimoon** [coreos/coreos-kubernetes#739]
kube-aws version: v0.8.3

CoreOS version:

```
NAME=CoreOS
ID=coreos
VERSION=1122.3.0
VERSION_ID=1122.3.0
BUILD_ID=2016-10-20-2045
PRETTY_NAME="CoreOS 1122.3.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

Kubernetes version: `version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.3+coreos.0", GitCommit:"7819c84f25e8c661321ee80d6b9fa5f4ff06676f", GitTreeState:"clean", BuildDate:"2016-10-17T21:19:17Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}`

Kubernetes Controller Instance type: `t2.small`
Kubernetes Worker Instance type: `m4.xlarge`

We ran into an issue where commands like `df -h` hang on `worker server`. We ran with `strace` and it hang at:

```
stat("/proc/sys/fs/binfmt_misc",
```

Rebooting worker instances will resolve the problem temporarily, but after about 1 day it will occur again.

**Comment by pbx0** [comment]
@keimoon Is `df -h` being run inside a pod/container or is it directly executed on the host? If it is a pod could you share the pod manifest?

**Comment by felixbuenemann** [comment]
@keimoon I had a workaround for this on CoreOS stable which consisted of the following drop-in:

```yaml
coreos:
  units:
    - name: docker.service
      drop-ins:
        - name: 60-mount-binfmt-misc.conf
          content: |
            [Unit]
            Wants=proc-sys-fs-binfmt_misc.mount
            After=proc-sys-fs-binfmt_misc.mount
```

My git commit message for that change said:

> Add workaround for binfmt_misc automount issues
>
> Without this fix storage monitoring in telegraf is not working because
the try to trigger the automount of /rootfs/proc/sys/fs/binfmt_misc
fails and causes a "Too many levels of symbolic links" error.

This fix is no longer working for me on CoreOS beta, so maybe I need to revert it because the underlying systemd issue was fixed.

**Comment by keimoon** [comment]
@pbx0 I ran `df -h` directly on the host.

**Comment by pbx0** [comment]
Just linking these two issues for the time being: https://github.com/coreos/bugs/issues/1630. Thanks for the extra info

**Comment by keimoon** [comment]
@lucab, @pbx0 so it happened again. Here is there output of `rkt list`:

```
UUID		APP		IMAGE NAME					STATE	CREATED		STARTED		NETWORKS
1b4320a3	flannel		quay.io/coreos/flannel:0.5.5			running	1 day ago	1 day ago
88f0d25c	hyperkube	quay.io/coreos/hyperkube:v1.4.3_coreos.0	running	1 day ago	1 day ago
d8ef9314	node		quay.io/calico/node:v0.19.0			running	1 day ago	1 day ago
```

Also, there are some failed units:

```
UNIT                                                                                                    LOAD   ACTIVE SUB    DESCRIPTION
● var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount      loaded failed failed /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-********
● var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount      loaded failed failed /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-********
● var-lib-kubelet-pods-290523a2\x2d9a70\x2d11e6\x2daa27\x2d0e129d5a939c-volumes-kubernetes.io\x7eaws\x2debs-pvc\x2d1a1e6ac5\x2d9a70\x2d11e6\x2daa27\x2d0e129d5a939c.mount loaded failed failed /var/lib/kubelet/pods/290523a2-9a70-11e6
● var-lib-kubelet-pods-36abc8ec\x2d99c5\x2d11e6\x2daa27\x2d0e129d5a939c-volumes-kubernetes.io\x7eaws\x2debs-pvc\x2d2f340719\x2d99c5\x2d11e6\x2daa27\x2d0e129d5a939c.mount loaded failed failed /var/lib/kubelet/pods/36abc8ec-99c5-11e6
● var-lib-rkt-pods-run-88f0d25c\x2deb89\x2d45c5\x2dbfb0\x2d8ec4dc6e534d-stage1-rootfs-opt-stage2-hyperkube-rootfs-var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2da9ec6e0f.mount loaded failed fail
● var-lib-rkt-pods-run-88f0d25c\x2deb89\x2d45c5\x2dbfb0\x2d8ec4dc6e534d-stage1-rootfs-opt-stage2-hyperkube-rootfs-var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2df7d95051.mount loaded failed fail
● locksmithd.service                                                                                      masked failed failed locksmithd.service
```
Output of `systemctl status [unit]` all look like this:

```
● var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount - /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-********
   Loaded: loaded (/proc/self/mountinfo)
   Active: failed (Result: resources)
    Where: /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-********
     What: /dev/xvdbd

Oct 25 05:02:01 ip-*** systemd[1]: var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount: Failed to run 'mount' task: No such file or directory
Oct 25 05:02:01 ip-*** systemd[1]: Failed to mount /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-********.
Oct 25 05:02:01 ip-*** systemd[1]: var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount: Unit entered failed state.
Oct 25 08:42:28 ip-*** systemd[1]: var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-us\x2deast\x2d1b-vol\x2d********.mount: Failed to run 'mount' task: No such file or directory
Oct 25 08:42:28 ip-*** systemd[1]: Failed to mount /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-1b/vol-*********.
```

Also, we are using persistent volume claim with EBS. 

This is our test cluster so we can keep the nodes in current state for further investigating (the kubernetes pods are still running fine though). Our production cluster (using coreos 1068.8.0 and kubernetes 1.4.0_coreos.2) does not have this error.

Thanks.

**Comment by keimoon** [comment]
So the `df -h` hanging issue is caused by `node_exporter` (https://github.com/prometheus/node_exporter) blocking `statfs` on `/var/lib/rkt/pods/run/d8ef9314-b65a-4d5a-9bf2-47fc3a48a587/stage1/rootfs/opt/stage2/node/rootfs/proc/sys/fs/binfmt_misc`. Stopping `node_exporter` will cure the problem of `df -h`, but I still cannot `ls /proc/sys/fs/binfmt_misc`.

Another thing is somehow all mount points for `binfmt_misc` is missing:

On node with problem:
```
mount | grep binfmt
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11628)
systemd-1 on /var/lib/rkt/pods/run/d8ef9314-b65a-4d5a-9bf2-47fc3a48a587/stage1/rootfs/opt/stage2/node/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11628)
systemd-1 on /var/lib/rkt/pods/run/88f0d25c-eb89-45c5-bfb0-8ec4dc6e534d/stage1/rootfs/opt/stage2/hyperkube/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11628)
systemd-1 on /var/lib/rkt/pods/run/1b4320a3-bd88-4445-8fa1-9973dc11350b/stage1/rootfs/opt/stage2/flannel/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11628)
systemd-1 on /var/lib/rkt/pods/run/88f0d25c-eb89-45c5-bfb0-8ec4dc6e534d/stage1/rootfs/opt/stage2/hyperkube/rootfs/var/lib/rkt/pods/run/1b4320a3-bd88-4445-8fa1-9973dc11350b/stage1/rootfs/opt/stage2/flannel/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=31,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11628)
```

While on node without problem:

```
mount | grep binfmt
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=16697)
systemd-1 on /var/lib/rkt/pods/run/495fe6f5-3eff-4eed-bfc0-eb4671d3780e/stage1/rootfs/opt/stage2/hyperkube/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=16697)
systemd-1 on /var/lib/rkt/pods/run/abe467ce-100d-4a3e-b0f1-b44de327e29c/stage1/rootfs/opt/stage2/node/rootfs/proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=16697)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
binfmt_misc on /var/lib/rkt/pods/run/495fe6f5-3eff-4eed-bfc0-eb4671d3780e/stage1/rootfs/opt/stage2/hyperkube/rootfs/proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
binfmt_misc on /var/lib/rkt/pods/run/abe467ce-100d-4a3e-b0f1-b44de327e29c/stage1/rootfs/opt/stage2/node/rootfs/proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
```

**Comment by colhom** [comment]
Hello Kubernetes Community,

Future work on kube-aws will be moved to [a new dedicated repository](https://github.com/coreos/kube-aws). @mumoshu will be running point on maintaining that repository- please move all issues and PRs over there as soon as you can. We will be halting active development on the AWS portion of this repository in the near future. We will continue to maintain the vagrant single and multi-node distributions in this repository, along with our [hyperkube container image](https://quay.io/coreos/hyperkube) image.

A community announcement to end users will be made once the transition is complete. We at CoreOS ask that those reading this message avoid publicizing/blogging about the transition until the official annoucement has been made to the community in the next week.

The new dedicated kube-aws repository already has the following features merged in:
* Discrete etcd cluster
* HA control plane
* Cluster upgrades
* Node draining/cordoning

If anyone in the Kubernetes community would like to be involved with maintaining this new repository, find @chom and/or @mumoshu on the Kubernetes slack in the #sig-aws channel or via direct message.

~CoreOS Infra Team

**Comment by euank** [comment]
Closing as a dupe of https://github.com/coreos/bugs/issues/1630 where there's also more information.
